src/persistence.rs

```diff
@@ -73,26 +80,70 @@ fn write_regressor_header(output_bufwriter: &mut dyn io::Write) -> Result<(), Bo
     Ok(())
 }
 
-fn load_regressor_without_weights(input_bufreader: &mut io::BufReader::<File>) 
-                        -> Result<(model_instance::ModelInstance,
-                                   vwmap::VwNamespaceMap,
-                                   regressor::Regressor,
-                                 ), Box<dyn Error>> {
-    verify_header(input_bufreader).expect("Regressor header error");    
-    let vw = vwmap::VwNamespaceMap::new_from_buf(input_bufreader).expect("Loading vwmap from regressor failed");
-    let mi = model_instance::ModelInstance::new_from_buf(input_bufreader).expect("Loading model instance from regressor failed");
+fn load_regressor_without_weights(
+    input_bufreader: &mut io::BufReader<File>,
+    cmd_arguments: Option<&clap::ArgMatches>,
+) -> Result<
+    (
+        model_instance::ModelInstance,
+        vwmap::VwNamespaceMap,
+        regressor::Regressor,
+    ),
+    Box<dyn Error>,
+> {
+    verify_header(input_bufreader).expect("Regressor header error");
+    let vw = vwmap::VwNamespaceMap::new_from_buf(input_bufreader)
+        .expect("Loading vwmap from regressor failed");
+
+    // Let's create a mutable model from the template and modify it based on CMDs
+    let mi: model_instance::ModelInstance = match cmd_arguments {
+        Some(p) => {
+            let mut mi = model_instance::ModelInstance::new_from_buf(input_bufreader)
+                .expect("Loading model instance from regressor failed");
+
+            // if hyperparam. is part of the cmd, try to replace it
+            for hyperparameter_value in ONLINE_HYPERPARAMETER_CANDIDATES.iter() {
```

Comment:
please read parameters in the same explicit way as it is done in model_instance.rs (and do it there, as the comment above says)
So we keep the consistency.

The reason is that this code is really confusing since you first have a list, but naturally you have to have exact match afterward anyway. So this is error prone. For example - iparameters in ONLINE_HYPERPARAMETER_CANDIDATES could easily not match string names in used in inner match here...
this can result in silent breakage.

Reply:
Refactored cmd handling to be aligned with the existing logic (`new_from_cmdline`); the loop-based matching is not needed anymore indeed